scripts/globals/effects/reprisal.lua

```diff
@@ -8,7 +8,7 @@ local effectObject = {}
 effectObject.onEffectGain = function(target, effect)
     target:addMod(xi.mod.SPIKES, 6)
     -- Spike damage is calculated on hit in battleutils::TakePhysicalDamage
-    target:setMod(xi.mod.SPIKES_DMG, 0)
+    target:setMod(xi.mod.SPIKES_DMG, effect:getPower())
```

Comment:
why was this needed?

Reply:
Spikes returned 0 damage regardless of what was being fought. Adding in the effect power returned damage values.

Reply:
Hmm. Smells like there was a PC specific check, perhaps. It should be calculated elsewhere. Can't look into it right now but will check it out

Reply:
I'm not sure, I went through the damage calculations for spike damage in Battleutils, tried a few variations of the subeffect exception but no luck with any of the adjustments. They are blocking and mitigation of incoming damage is there, even the reprisal animations but there's no spike damage processing.

Reply:
`Action->spikesParam;` is where the spikes damage normally lived. this feeds into `CalculateSpikeDamage()` where modifications happen.

Reply:
no param value, no dmg

Reply:
https://github.com/LandSandBoat/server/blob/dcb1b856a7166a99960a4efbcca7b9b4c8b36c14/src/map/utils/battleutils.cpp#L792

Looks like the math is pulling calculations off a gear slot... may need a trust exception for the main slot. Or I'm reading this entirely wrong. 😜

Reply:
https://github.com/LandSandBoat/server/blob/082fcf03de3992125475c9bd1af35a1047ea909e/src/map/utils/battleutils.cpp#L2065
probably need a trust exception for a trust with shield, this is also where reprisal spikes dmg is being calculated

Reply:
I can take a swing at revising this to allow PET and TRUSTs to bypass the equipment check which should make the mod work for them and apply phantom blocks.